spec/allocator_spec.rb

```diff
@@ -53,26 +53,99 @@
       expect { new_allocator(1).allocate([rate, 1 - rate]) }.not_to raise_error
     end
 
-    specify "#allocate fills pennies from beginning to end with roundrobin strategy" do
-      moneys = new_allocator(0.05).allocate([0.3,0.7], :roundrobin)
-      expect(moneys[0]).to eq(Money.new(0.02))
-      expect(moneys[1]).to eq(Money.new(0.03))
+    specify "#allocate raise ArgumentError when invalid strategy is provided" do
+      expect { new_allocator(0.03).allocate([0.5, 0.5], :bad_strategy_name) }.to raise_error(ArgumentError, "Invalid strategy. Valid options: :roundrobin, :roundrobin_reverse, :nearest")
     end
 
-    specify "#allocate fills pennies from end to beginning with roundrobin_reverse strategy" do
-      moneys = new_allocator(0.05).allocate([0.3,0.7], :roundrobin_reverse)
-      expect(moneys[0]).to eq(Money.new(0.01))
-      expect(moneys[1]).to eq(Money.new(0.04))
+    specify "#allocate raises an error when splits exceed 100%" do
+      expect { new_allocator(0.03).allocate([0.5, 0.6]) }.to raise_error(ArgumentError, "splits add to more than 100%")
```

Comment:
@mkorostoff-shopify are there tests in place where the split is under 100% ?

Does it work / how does it behave?

does [0.3,0.3] == [0.5,0.5]

I am currently wondering if _this_ could be when left_over.length might be longer than amounts.length 🤔

Reply:
Got it, thanks :+1:

Feel free to add a test if you think it's valuable.

I would also like to see some tests for non subunits=>decimal currencies (i.e. JPY which has no decimals), just in case 🤔

Reply:
@raphpap thank you for this feedback, I have implemented the tests you suggested for yen allocation as well as allocations adding to less than 100% in my latest commit, 803cb3f